core/src/main/java/org/apache/druid/guice/GuiceAnnotationIntrospector.java

```diff
@@ -58,9 +58,9 @@ public Object findInjectableValueId(AnnotatedMember m)
       if (m instanceof AnnotatedMethod) {
         throw new IAE("Annotated methods don't work very well yet...");
       }
-      return Key.get(m.getGenericType());
+      return Key.get(m.getRawType());
```

Comment:
This change confuses me. The javadoc of `getGenericType` says that `getType` should be used to replace it. But according to the CI test result of #12373 , using of `getType` here seems that it does not work correctly.